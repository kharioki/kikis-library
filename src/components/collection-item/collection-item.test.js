import React from "react";
import { shallow } from "enzyme";

import CollectionItem from "./collection-item.component";
import CustomButton from "../custom-button/custom-button.component";

describe("CollectionItem tests", () => {
  let wrapper;
  let mockAdd;
  const imageUrl = "www.someimage.com";
  const name = "some image";

  beforeEach(() => {
    mockAdd = jest.fn();

    const mockProps = {
      item: {
        name: name,
        imageUrl: imageUrl
      },
      addItem: mockAdd
    };

    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it("should render CollectionItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call addItem when CustomButton clicked", () => {
    wrapper.find(CustomButton).simulate("click");
    expect(mockAdd).toHaveBeenCalled();
  });
});
