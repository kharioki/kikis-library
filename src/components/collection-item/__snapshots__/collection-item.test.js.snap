// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem tests should render CollectionItem component 1`] = `ShallowWrapper {}`;
